src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -96,157 +120,257 @@ public void Stop()
         }
 
         /// <inheritdoc />
-        public virtual void ProcessBlock(Block block)
+        public void ProcessBlock(Block block)
         {
-            Guard.NotNull(block, nameof(block));
+            this.logger.LogTrace("()");
+
             this.logger.LogTrace("({0}:'{1}')", nameof(block), block.GetHash());
 
-            ChainedHeader newTip = this.chain.GetBlock(block.GetHash());
-            if (newTip == null)
-            {
-                this.logger.LogTrace("(-)[NEW_TIP_REORG]");
-                return;
-            }
+            this.QueueBlock(block);
+
+            this.logger.LogTrace("(-)");
+        }
+
+        /// <inheritdoc />
+        public virtual void ProcessTransaction(Transaction transaction)
+        {
+            this.logger.LogTrace("()");
+
+            Guard.NotNull(transaction, nameof(transaction));
+
+            this.logger.LogTrace("({0}:'{1}')", nameof(transaction), transaction.GetHash());
+
+            this.walletManager.ProcessTransaction(transaction);
+
+            this.logger.LogTrace("(-)");
+        }
+
+        /// <inheritdoc />
+        public virtual void SyncFromDate(DateTime date)
+        {
+            this.logger.LogTrace("()");
```

Comment:
You don't need this line of log.

Reply:
Also can you please move SyncFromDate and SyncFromHeight back down to the end of the class? It's just that it'll be easier for us to compare the changes. Thanks 😗

Reply:
They moved as I wanted to keep all private methods at the bottom.  They are back at the bottom of the class now.